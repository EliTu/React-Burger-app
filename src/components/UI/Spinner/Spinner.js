import React from 'react';
import styles from './Spinner.module.css';

const Spinner = props => {
	// CSS Modules styles:
	const { Loader } = styles;

	return <div className={Loader}>Loading...</div>;
};

export default Spinner;
