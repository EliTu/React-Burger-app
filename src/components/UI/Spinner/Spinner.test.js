import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './Spinner';

describe('<Spinner/>', () => {
	let component;
	beforeEach(() => {
		component = shallow(<Spinner />);
	});

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component).toBeTruthy();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should have a div with loading text', () => {
		expect(component.find('.Loader').text()).toEqual('Loading...');
	});
});
