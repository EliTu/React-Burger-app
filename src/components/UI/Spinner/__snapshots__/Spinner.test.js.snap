// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> should render without errors 1`] = `
<div
  className="Loader"
>
  Loading...
</div>
`;
